service/src/main/java/uk/gov/hmcts/reform/fpl/service/DocmosisDocumentGeneratorService.java

```diff
@@ -57,4 +62,15 @@ public DocmosisDocument generateDocmosisDocument(Map<String, Object> templateDat
 
         return new DocmosisDocument(docmosisTemplate.getDocumentTitle(), response);
     }
+
+    public String generateDraftWatermarkEncodedString() {
```

Comment:
I doubt it is good place for it. It adds unnecessary responsibility to the service that was focus on one concrete thing so far. For me there should be separate class that reads assets and encodes them in B64.

Reply:
We initially had it in a different class but decided to bring in here with the idea that docmosis related stuff be in the same service. But your suggestion is right. We will bring the new service class

Reply:
Moved to a new abstract class that future docmosis template data providers can then extend